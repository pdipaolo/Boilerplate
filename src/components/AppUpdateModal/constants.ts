const APP_UPDATE_TYPE = {
  SOFT: 1 as 1,
  HARD: 2 as 2,
};

export { APP_UPDATE_TYPE };
