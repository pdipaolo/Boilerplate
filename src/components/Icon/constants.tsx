import React from 'react';

import { type SvgProps } from 'react-native-svg';

import Bonus from '../../assets/svg/bonus.svg';

const ICON = {
  // General
  BONUS: 'bonus' as 'bonus',
};

const ICON_COMPONENT = {
  // General
  BONUS: {
    ID: ICON.BONUS,
    COMPONENT: (props: SvgProps) => <Bonus {...props} />,
  },
};

function findIconById(
  id: (typeof ICON)[keyof typeof ICON] | undefined,
  props: SvgProps & {
    primaryColor: string;
    secondaryColor?: string;
    tertiaryColor?: string;
  },
): JSX.Element | undefined {
  const ICONS = Object.values(ICON_COMPONENT);
  const CURR_ICON = ICONS.find((obj) => obj.ID === id);
  return CURR_ICON?.COMPONENT(props);
}

export { ICON, findIconById };
