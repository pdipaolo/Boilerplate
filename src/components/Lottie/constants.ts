import LoaderLottie from '../../assets/animations/loader.json';

const ANIMATION = {
  LOADER: 'loader' as 'loader',
};

const ANIMATION_DATA = {
  [ANIMATION.LOADER]: LoaderLottie,
};

export { ANIMATION, ANIMATION_DATA };
