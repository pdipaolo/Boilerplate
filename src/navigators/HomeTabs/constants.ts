import tab1Route from '../../screens/Tab1/route';
import tab2Route from '../../screens/Tab2/route';
import tab3Route from '../../screens/Tab3/route';
import tab4Route from '../../screens/Tab4/route';
import tab5Route from '../../screens/Tab5/route';

const bottomRoutes = [tab1Route, tab2Route, tab3Route, tab4Route, tab5Route];

export { bottomRoutes};
