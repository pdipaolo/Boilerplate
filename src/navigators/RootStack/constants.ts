import createAccountRoute from '../../screens/CreateAccount/route';
import loginRoute from '../../screens/Login/route';

const routes = [loginRoute, createAccountRoute];

export { routes };
