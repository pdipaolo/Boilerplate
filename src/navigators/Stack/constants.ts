import { createStackNavigator } from '@react-navigation/stack';

import bottomTabsRoute from '../../screens/BottomTabs/route';
import introRoute from '../../screens/Intro/route';

import { type StackParamsList } from './types';

const Stack = createStackNavigator<StackParamsList>();
const StackNavigator = Stack.Navigator;
const StackScreen = Stack.Screen;

const routes = [
  introRoute,
  bottomTabsRoute,
];

export { StackNavigator, StackScreen, routes };
