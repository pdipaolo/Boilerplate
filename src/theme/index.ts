import { palette } from './colors';
import { typography } from './typography';

const theme = {
  typography,
  palette,
};

export default theme;
