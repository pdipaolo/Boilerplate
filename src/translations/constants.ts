const IT: 'it' = 'it';
const EN: 'en' = 'en';

const LANGUAGES = {
  IT,
  EN,
};

const supportedLanguages: Array<typeof LANGUAGES.EN | typeof LANGUAGES.IT> = [
  LANGUAGES.EN,
  LANGUAGES.IT,
];

const DEFAULT_LANGUAGE = LANGUAGES.IT;

export {LANGUAGES, supportedLanguages, DEFAULT_LANGUAGE};
