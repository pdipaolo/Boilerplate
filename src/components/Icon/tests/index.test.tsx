import React from 'react';

import { render, screen } from '@testing-library/react-native';

import Icon from '../';
import theme from '../../../theme';
import { ICON } from '../constants';

describe('Icon', () => {
  it('should match snapshot', () => {
    render(<Icon src="https://placehold.co/600x400" />);

    expect(screen).toMatchSnapshot();
  });

  it('should render as link with width: 600', () => {
    const width = 600;

    const { root } = render(
      <Icon src="https://placehold.co/600x400" width={width} />,
    );

    const received = root.props.children.props.width;
    const expected = width;

    expect(received).toBe(expected);
  });

  it('should render as link with minHeight: 24', () => {
    const { root } = render(<Icon src="https://placehold.co/600x400" />);

    const received = root.props.style.minHeight;
    const expected = 24;

    expect(received).toBe(expected);
  });

  it('should render as id with width: 600', () => {
    const width = 600;

    const { root } = render(<Icon id={ICON.USER_CIRCLE} width={width} />);

    const received = root.props.width;
    const expected = width;

    expect(received).toBe(expected);
  });

  it(`should render as id with primaryColor: ${theme.palette.grey[900]} as default`, () => {
    const { root } = render(<Icon id={ICON.USER_CIRCLE} />);

    const received = root.props.primaryColor;
    const expected = theme.palette.grey[900];

    expect(received).toBe(expected);
  });

  it(`should render as id with primaryColor: ${theme.palette.primary.main} when prop passed`, () => {
    const { root } = render(
      <Icon id={ICON.USER_CIRCLE} primaryColor={theme.palette.primary.main} />,
    );

    const received = root.props.primaryColor;
    const expected = theme.palette.primary.main;

    expect(received).toBe(expected);
  });
});
