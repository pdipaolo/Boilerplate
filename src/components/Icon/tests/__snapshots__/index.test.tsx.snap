// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot 1`] = `
<View
  style={
    {
      "minHeight": 24,
    }
  }
/>
`;
